base_optional_quick_create/model.py

```diff
@@ -31,35 +31,21 @@ class IrModel(orm.Model):
         }
 
     def _patch_quick_create(self, cr, ids):
-
-        def _wrap_name_create():
-            def wrapper(self, cr, uid, name, context=None):
+        def _wrap_name_create(self, cr, uid, name, context=None):
+            if self.pool.get(self._name).avoid_quick_create:
```

Comment:
Thank you @Yajo 

but ```avoid_quick_create``` is not a field of the patched model: it is a field of ```ir.model```.

So, I tried with
```
    def _patch_quick_create(self, cr, ids):
        def _wrap_name_create(self, cr, uid, name, context=None):
            model_pool = self.pool['ir.model']
            model_record_ids = model_pool.search(cr, uid, [
                ('model', '=', self._name),
                ], context=context)
            model_record = model_pool.browse(
                cr, uid, model_record_ids[0], context)
            if model_record.avoid_quick_create:
                raise orm.except_orm(
                    _('Error'),
                    _("Can't create quickly. Opening create form"))
            else:
                return _wrap_name_create.origin(
                    self, cr, uid, name, context=context)

        for model in self.browse(cr, SUPERUSER_ID, ids):
            self.pool.get(model.model)._patch_method(
                'name_create', _wrap_name_create)

        return True
```

but got

```
server-tools/base_optional_quick_create/model.py", line 47, in _wrap_name_create
    self, cr, uid, name, context=context)
TypeError: unbound method name_create() must be called with workflow.workitem instance as first argument (got res.partner.title instance instead)
```